lib/tasks/tranche_maker.rake

```diff
@@ -13,17 +13,18 @@ namespace :tranche do
     beta_test_users = YAML.load_file(Rails.root.join('config/encrypted_private_beta_users.yml'))
 
     new_users.each do |user|
-      encoded_user = user.upcase.tr(' ', '')
+      encoded_user = user.upcase.gsub(' ', '%20')
       uri = URI("https://ccms-pda.legalservices.gov.uk/api/providerDetails/#{encoded_user}")
+      puts uri.to_s
       response = Net::HTTP.get(uri)
       hash = JSON.parse(response)
       if hash.key?('error')
         puts ">>>>>>>> User #{user} / #{encoded_user} not found <<<<<<<<<"
       else
-        contact_hash = hash['contacts'].detect { |h| h['name'] == encoded_user.upcase }
-        whitelisted_users << encoded_user
+        contact_hash = hash['contacts'].detect { |h| h['name'] == user.upcase }
+        whitelisted_users << user
```

Comment:
Does this mean that a user with a space in their name will be in the whitelist as 'TEST USER' but will be in the portal as TEST%20USER?

Will they match on login?

Reply:
Exactly.  Take a look at the user that was causing all this trouble: RUFIA ADALAT.

He or she logs in apparently as `RUFIA ADELAT`.  We have to look him/her up as `RUFIA%20ADELAT`.  The name in the list of contacts is retrurned as `RUFIA ADELAT`.  That all makes sense actually if you're dumb enough to allow spaces in user names.